### PR TITLE
toc scrolls smoothly and is collapsible

### PR DESCRIPTION
**Pull request by SiriusXT**
1. Notes (including read-only and edit mode) scroll smoothly when clicking toc
2. After clicking to close the toc, display a button to allow the user to open the toc again

### CODE REVIEW

**Comment by zadam** [comment]
The collapsed TOC looks strange:

image

**Comment by SiriusXT** [comment]
Made some changes: After clicking to close the toc, display a button to allow the user to open the toc again
图片

File: src/public/app/widgets/toc.js [hunk 1]
```diff
@@ -244,9 +249,29 @@ export default class TocWidget extends RightPanelWidget {
...
+        var $noteContainer, $noteSplit;
+        if (isReadOnly) {
+            $noteContainer = await this.noteContext.getContentElement();
+
+        } else {
+            const textEditor = await this.noteContext.getTextEditor();
+            $noteContainer = $(textEditor.editing.view.domRoots.values().next().value);
+        }
+        $noteSplit = $noteContainer.closest('.component.note-split.type-text.mime-text-html');
+        $noteSplit.find(".ribbon-button-container").append(`<button type="button" class="icon-action bx bx-menu-alt-right show-toc" title="Show Toc"></button>`);
```
Comment (zadam) [inline src/public/app/widgets/toc.js:266]: This is rather hacky, you can check e.g. `edit_button.js` on how to do this properly ...

Comment (SiriusXT) [reply src/public/app/widgets/toc.js:266]: I created a new show_toc_button.js under buttons, but I don't know how to call this.triggerEvent('reEvaluateRightPaneVisibility'); in this file.
图片
图片
click no response

Comment (zadam) [reply src/public/app/widgets/toc.js:266]: That command works only within the sidebar, but I could fix that and you don't have to worry about it.

**Pull request closed without merging**